File: backend/app/models/mention.py
"""Mention data models for tracked brand references."""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from enum import Enum


class SentimentScore(str, Enum):
    POSITIVE = "positive"
    NEUTRAL = "neutral"
    NEGATIVE = "negative"
    CRITICAL = "critical"


class MentionCreate(BaseModel):
    """Data for a new brand mention."""
    brand_id: str
    source_url: str
    content: str
    platform: str
    author: Optional[str] = None


class Mention(MentionCreate):
    """A tracked brand mention."""
    id: str
    detected_at: datetime
    sentiment: SentimentScore = SentimentScore.NEUTRAL
    compliance_score: Optional[float] = None
    threat_level: Optional[str] = None
    scout_id: Optional[str] = None

File: frontend/src/types/mention.ts
/** Sentiment categories for brand mentions */
export type Sentiment = "positive" | "neutral" | "negative" | "critical";

/** A tracked brand mention */
export interface Mention {
  id: string;
  brand_id: string;
  source_url: string;
  content: string;
  platform: string;
  author?: string;
  detected_at: string;
  sentiment: Sentiment;
  compliance_score?: number;
  threat_level?: string;
  scout_id?: string;
}
